### PR TITLE
Todo Items

### PR DESCRIPTION
**Pull request by loicteixeira**
User can create todo items

#### User Stories
- As a user, I want to create multiple to-do items

#### Required Tests
- Are you able to create multiple to-do items?
- Are to-do items displayed after creation?

### CODE REVIEW

File: app/controllers/items_controller.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    @items = current_user.items.all
+    @item = Item.new
+  end
+
+  def create
+    @item = current_user.items.build(item_params)
+    if @item.save
+      redirect_to items_path
+    else
+      render :index
```
Comment (eliotsykes) [inline app/controllers/items_controller.rb]: Have you got any thoughts on showing the user info about failed saves?

File: spec/models/item_spec.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    let(:user) { create(:user) }
+    let(:item) { create(:item, user: user)}
+
+    it 'should be valid' do
+      expect(item).to be_valid
+    end
+
+    it 'should require a name' do
+      item.name = ''
+      expect(item).to be_invalid
```
Comment (eliotsykes) [inline spec/models/item_spec.rb]: Consider verifying the `item.errors[:name]` message is as expected so you can be sure the correct validation error is being triggered. 

If you do add that expectation, afterward you might want to take a look at shoulda-matchers and its handy matchers. https://github.com/thoughtbot/shoulda-matchers

**Comment by eliotsykes** [comment]
This is in very good shape :+1:

File: app/controllers/items_controller.rb (hunk 1 above)
Comment (loicteixeira) [inline app/controllers/items_controller.rb]: Definitely. However I'm not sure how I want it to display (inline or flash message) so I'll come back to it after the first design draft.

Comment (eliotsykes) [inline app/controllers/items_controller.rb]: :+1:

File: spec/models/item_spec.rb (hunk 2 above)
Comment (loicteixeira) [inline spec/models/item_spec.rb]: I've added another expectation in 70837a47af5b9da0e92b5757a931dcc0991e4dda but it doesn't feel quite right.

File: app/controllers/items_controller.rb (hunk 1 above)
Comment (loicteixeira) [inline app/controllers/items_controller.rb]: Added in b74a32cfd6bf268076969aa9847708f103d73cd8. This might change though.

**Comment by loicteixeira** [comment]
1. `expect(item.errors[:name][0]).to match /can't be blank/` tests that *the error message include this substring and is the first error in the list (it does not care of other error messages as long as they come after this one).*
1. `expect(item.errors[:name]).to eq ["can't be blank"]` tests that *this specific error message exists and there is no other error messages.*
1. `expect(item.errors[:name]).to include "can't be blank"` test that *this specific error message exists (it does not care of other error messages at all).*

1 and 2 are too specific, they test more than one thing, although 1 allow substring searches which can be useful. Therefore, I think I'll favour 3 in the future, unless the order is important or you have to be sure no other error message is being produced.

**Comment by loicteixeira** [comment]
1. `expect(item.errors[:name][0]).to match /can't be blank/` tests that *the error message include this substring and is the first error in the list (it does not care of other error messages as long as they come after this one).*
1. `expect(item.errors[:name]).to eq ["can't be blank"]` tests that *this specific error message exists and there is no other error messages.*
1. `expect(item.errors[:name]).to include "can't be blank"` test that *this specific error message exists (it does not care of other error messages at all).*

1 and 2 are too specific, they test more than one thing, although 1 allow substring searches which can be useful. Therefore, I think I'll favour 3 in the future, unless the order is important or you have to be sure no other error message is being produced.

**Pull request merged**